### PR TITLE
Add new API methods for managing ISV cloud provider credentials

### PR DESCRIPTION
**Pull request by WangLiNaruto**

### CODE REVIEW

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/IsvCloudCredentialsApi.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+     * @return Returns list the available credential types of the cloud service provider.
+     */
+    @Tag(name = "Credentials Management",
+            description = "APIs for managing isv's cloud provider credentials")
+    @GetMapping(value = "/credential_types",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(description = "Users in the ISV role get the list of available credential types of"
+            + " the cloud service provider.")
+    public List<CredentialType> getIsvCloudCredentialTypes(
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/IsvCloudCredentialsApi.java:71]: This is a point that I missed to mention. This method and the next method (getIsvCloudCredentialCapabilities) are common to both ISV and end users. 
We can move it to another common API controller - CredentialsConfigApi. The definition of cloud credentials will be at just one place and will be used by both ISV and end users.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/common/exceptions/UserNoLoginException.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.modules.models.common.exceptions;
+
+/**
+ * Exception is thrown when the information of the currently logged in person cannot be found.
+ */
+public class UserNoLoginException extends  RuntimeException {
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/common/exceptions/UserNoLoginException.java:11]: Should be UserNotLoggedInException

File: modules/api/src/main/java/org/eclipse/xpanse/api/exceptions/handler/CommonExceptionHandler.java [hunk 3]
```diff
@@ -170,4 +171,16 @@ public Response handleMethodArgumentTypeMismatchException(
...
                 Collections.singletonList(ex.getMessage()));
     }
+
+    /**
+     * Exception handler for UserNoLoginException.
+     */
+    @ExceptionHandler({UserNoLoginException.class})
+    @ResponseStatus(HttpStatus.UNAUTHORIZED)
+    @ResponseBody
+    public Response handleUserNoLoginException(
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/exceptions/handler/CommonExceptionHandler.java:181]: please also add a test for this.

File: modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java [hunk 4]
```diff
@@ -117,9 +117,6 @@ public String getCredentialOpenApiUrl(Csp csp, CredentialType type) {
      */
     public List<AbstractCredentialInfo> listCredentials(Csp csp, CredentialType type,
```
Comment (swaroopar) [inline modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java:118]: can you please confirm if the credential cache really works correctly when the userID is null?

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java (hunk 4 above)
Comment (WangLiNaruto) [reply modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java:118]: I've looked at the code and I think it works fine when userId is null
image
image
image

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**